src/main/java/org/datadog/jenkins/plugins/datadog/clients/ConcurrentMetricCounters.java

```diff
@@ -2,13 +2,31 @@
 
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.locks.ReentrantLock;
 
 public class ConcurrentMetricCounters {
-    private static final ConcurrentMap<CounterMetric, Integer> metrics = new ConcurrentHashMap<>();
-    public static final ThreadLocal<ConcurrentMap<CounterMetric, Integer>> Counters =
-            new ThreadLocal<ConcurrentMap<CounterMetric, Integer>>() {
-                @Override protected ConcurrentMap<CounterMetric, Integer> initialValue() {
-                    return metrics;
+    private static ConcurrentMetricCounters instance;
+    private static ConcurrentMap<CounterMetric, Integer> metrics = new ConcurrentHashMap<>();
```

Comment:
Should this be a static attribute since it's not used in static methods ?

Reply:
Since it is a singleton it doesn't really matter I believe. It should enforce the fact that we should only have one instance per JVM.